### PR TITLE
feat(new integration): Onboarding Engage Device Mode

### PR DESCRIPTION
**Pull request by anantjain45823**
## Description of the change

> Onboarding new destination Engage

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

- - -
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/662)

### CODE REVIEW

File: integrations/Engage/browser.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    const { message } = rudderElement;
+    let { userId, originalTimestamp } = message;
+    if (!userId) {
+      userId = get(message, "context.traits.userId");
+      if (!userId) {
+        logger.error("userId is required for Identify call.");
+        return;
+      }
+    }
+    const number = get(message, "context.traits.phone");
```
Comment (shrouti1507) [inline integrations/Engage/browser.js:73]: do not use get and set in device mode.

File: integrations/Engage/browser.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  }
+
+  track(rudderElement) {
+    logger.debug("===In Engage track===");
+    const { message } = rudderElement;
+    const { event, properties, originalTimestamp, userId } = message;
+    if (!userId) {
+      userId = get(message, "context.traits.userId");
+      if (!userId) {
+        logger.error("userId is required for Identify call.");
```
Comment (shrouti1507) [inline integrations/Engage/browser.js:102]: this is track call. Edit the comment.

File: integrations/Engage/browser.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+  }
+
+  page(rudderElement) {
+    logger.debug("===In Engage page===");
+    const { message } = rudderElement;
+    const { name, properties, originalTimestamp, category, userId } = message;
+    if (!userId) {
+      userId = get(message, "context.traits.userId");
+      if (!userId) {
+        logger.error("userId is required for Identify call.");
```
Comment (shrouti1507) [inline integrations/Engage/browser.js:127]: edit the statement, this is page

File: integrations/Engage/browser.js [hunk 4]
```diff
@@ -62,23 +62,12 @@ class Engage {
     logger.debug("===In Engage identify");
 
     const { message } = rudderElement;
-    let { userId, originalTimestamp } = message;
+    const { userId, fName, number, lName } = getDefinedTraits(message);
```
Comment (shrouti1507) [inline integrations/Engage/browser.js:65]: for userId align with cloud mode which is:
 const uid =
    getDestinationExternalID(message, "engageId") ||
    getFieldValueFromMessage(message, "userIdOnly");

File: integrations/Engage/browser.js [hunk 5]
```diff
@@ -95,13 +84,12 @@ class Engage {
   track(rudderElement) {
     logger.debug("===In Engage track===");
     const { message } = rudderElement;
-    const { event, properties, originalTimestamp, userId } = message;
+    const { event, properties, originalTimestamp } = message;
+    const { userId } = getDefinedTraits(message);
```
Comment (shrouti1507) [inline integrations/Engage/browser.js:88]: const uid =
    getDestinationExternalID(message, "engageId") ||
    getFieldValueFromMessage(message, "userIdOnly");

File: integrations/Engage/browser.js [hunk 6]
```diff
@@ -120,13 +108,12 @@ class Engage {
   page(rudderElement) {
     logger.debug("===In Engage page===");
     const { message } = rudderElement;
-    const { name, properties, originalTimestamp, category, userId } = message;
+    const { name, properties, originalTimestamp, category } = message;
+    const { userId } = getDefinedTraits(message);
```
Comment (shrouti1507) [inline integrations/Engage/browser.js:112]: const uid =
    getDestinationExternalID(message, "engageId") ||
    getFieldValueFromMessage(message, "userIdOnly");

File: integrations/Engage/browser.js (hunk 4 above)
Comment (anantjain45823) [reply integrations/Engage/browser.js]: Okay making a new one in `getDefinedTraits` for `userIdOnly`

**Review by shrouti1507 — DISMISSED** [review]

**Review by Gauravudia — DISMISSED** [review]

**Review by MoumitaM — DISMISSED** [review]

**Review by MoumitaM — APPROVED** [review]

**Review by Gauravudia — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Pull request merged**